src/dnaio/_core.pyx

```diff
@@ -148,6 +148,24 @@ cdef class Sequence:
         """
         return self.fastq_bytes(two_headers=True)
 
+    def is_mate(self, Sequence other):
+        cdef:
+            char * header1_chars = NULL
+            char * header2_chars = NULL
+            size_t header1_length
+        # No need to check if type is unicode as it is guaranteed by the type.
+        if PyUnicode_KIND(self.name) == PyUnicode_1BYTE_KIND:
+            header1_chars = <char *>PyUnicode_1BYTE_DATA(self.name)
+            header1_length = <size_t> PyUnicode_GET_LENGTH(self.name)
+        else:
+            raise ValueError(f"name should be ASCII-only. Got {self.name:R}.")
```

Comment:
I think this line isn’t covered by tests. Can you add one? I think the `:R` won’t work (or will it?). Did you mean `!r` instead?

Reply:
It does not, you are right. I meant !r. (For float formatting `:` is used so Python has me confused now.)